src/thor/triplegbuilder.cc

```diff
@@ -1135,6 +1135,7 @@ void TripLegBuilder::Build(
   auto* tp_orig = trip_path.mutable_location(0);
   auto* tp_dest = trip_path.mutable_location(trip_path.location_size() - 1);
 
+  // TODO: use TimeInfo
```

Comment:
for the purpose of map matching we redo the calculation of time progression over the route, we could and should use TimeInfo here. any thoughts on whether it shouild be in this PR or the next? i am leaning towards this PR but its already getting large

Reply:
I'd say a new PR. Generally I'd like a PR to be mostly feature complete once opened. It quickly becomes hard for reviewers to follow along on open PRs that are also developed against.